### DISCUSSION TITLE
Blade/Alpine component is one state behind.

### DISCUSSION THREAD

**Discussion by robsontenorio** [livewire/livewire#8329] [1. Help]
Hey guys!

Just noticed there is a state "lag" between Livewire and Blade/Alpine component. Is there any known issue? Am I missing something?

  - Type "c" and see the issue regarding the size/count.

<hr />
<hr />

<img width="553" alt="image" src="https://github.com/livewire/livewire/assets/118955/55b8ede7-2303-4e54-881d-19ac7c5bbb0e">

<hr />
<hr />

```php
<?php

use App\Models\Tag;
use App\Models\User;
use Illuminate\Support\Collection;
use Livewire\Volt\Component;

new class extends Component {
    public string $search = '';

    public array $allLangs = ['html', 'css', 'javascript'];

    public function with(): array
    {
        return [
            'langs' => collect($this->allLangs)->filter(fn($lang) => str_contains($lang, $this->search))->all()
        ];
    }
}; ?>

<div>
    <div class="font-black">This component itself</div>
    <input wire:model.live="search" class="input input-primary" />
    @dump($langs)

    <div class="font-black mt-10">Blade Component</div>
    <x-my-component :options="$langs" />
</div>

```

```php
<?php

use Closure;
use Illuminate\Contracts\View\View;
use Illuminate\View\Component;

class MyComponent extends Component
{
    public function __construct(public $options = [])
    {
    }

    public function render(): View|Closure|string
    {
        return <<<'HTML'
                <div x-data="{ options: @js($options) }">
                        <div class="bg-green-50 p-5">
                            IT IS OK (blade variable): {{ count($options) }}
                        </div>

                        <div class="p-5 bg-red-100">
                            PROBLEM (alpine variable) <span x-text="options.length"></span>
                        </div>
                </div>
            HTML;
    }
}
```

**Comment by rpungello** [comment]
Did anything ever come of this? I saw a link [here](https://github.com/robsontenorio/mary/issues/368) and believe I'm having a similar issue, just with a single-select choices element instead of multiple.

**Reply by joshhanley** [reply]
The issue is that Alpine isn't designed to detected the changes from `options: @js($options)`. Instead you should be making use of `$wire.options` directly (in the example in this issue).

**Reply by rpungello** [reply]
Might be an issue within the MaryUI package then, because I was just using the same component from that as OP in the issue I linked to. Though as you've helped me figure out the issue I was having with FluxUI, I should be able to use that instead of the MaryUI choices component.
